reports/hosted-services/main.go

```diff
@@ -0,0 +1,199 @@
+	hoodawApiKey   = flag.String("hoodawAPIKey", os.Getenv("HOODAW_API_KEY"), "API key to post data to the 'How out of date are we' API")
+	hoodawEndpoint = flag.String("hoodawEndpoint", "/hosted_services", "Endpoint to send the data to")
+	hoodawHost     = flag.String("hoodawHost", os.Getenv("HOODAW_HOST"), "Hostname of the 'How out of date are we' API")
+	kubeconfig     = flag.String("kubeconfig", "kubeconfig", "Name of kubeconfig file in S3 bucket")
+	region         = flag.String("region", os.Getenv("AWS_REGION"), "AWS Region")
+
+	endPoint = *hoodawHost + *hoodawEndpoint
+)
+
+func main() {
+	flag.Parse()
+
+	// Gain access to a Kubernetes cluster using a config file stored in an S3 bucket.
+	clientset, err := authenticate.CreateClientFromS3Bucket(*bucket, *kubeconfig, *region, *ctx)
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	namespaces, err := GetNamespaces(clientset)
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	//make namespace map
+	nsDetailsMap := make(map[string]namespace, 0)
+
+	for _, ns := range namespaces {
+
+		namespaceDetails := GetNamespaceDetails(ns)
+		nsDetailsMap[namespaceDetails.Name] = namespaceDetails
+	}
+
+	// Get all ingress resources
+	ingressList, err := GetAllIngresses(clientset)
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	// Build the ingresses Map
+	ingressesMap := BuildIngressesMap(ingressList.Items)
+
+	// Add the ingress slice to the namespace Details map for all namespaces
+	for k, v := range nsDetailsMap {
+		if _, exist := ingressesMap[v.Name]; exist {
+			v.DomainNames = ingressesMap[v.Name]
+		}
+		nsDetailsMap[k] = v
+	}
+
+	jsonToPost, err := BuildJsonMap(nsDetailsMap)
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	// Post json to hoowdaw api
+	err = hoodaw.PostToApi(jsonToPost, hoodawApiKey, &endPoint)
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+}
+
+// GetNamespaces takes a Kubernetes clientset and returns all namespaces with type *v1beta1.IngressList and an error.
+
+func GetNamespaceDetails(ns v1.Namespace) namespace {
+
+	var namespaceDetails namespace
+
+	namespaceDetails.Name = ns.Name
+	namespaceDetails.Application = ns.Annotations["cloud-platform.justice.gov.uk/application"]
+	namespaceDetails.BusinessUnit = ns.Annotations["cloud-platform.justice.gov.uk/business-unit"]
+	namespaceDetails.DeploymentType = ns.Labels["cloud-platform.justice.gov.uk/environment-name"]
+	namespaceDetails.GithubURL = ns.Annotations["cloud-platform.justice.gov.uk/source-code"]
+	namespaceDetails.TeamName = ns.Annotations["cloud-platform.justice.gov.uk/team-name"]
+	namespaceDetails.TeamSlackChannel = ns.Annotations["cloud-platform.justice.gov.uk/slack-channel"]
+	namespaceDetails.DomainNames = []string{}
+	return namespaceDetails
+}
+
+// GetNamespaces takes a Kubernetes clientset and returns all namespaces with type *v1beta1.IngressList and an error.s
+func GetNamespaces(clientSet *kubernetes.Clientset) ([]v1.Namespace, error) {
```

Comment:
Could you use this instead:
https://github.com/ministryofjustice/cloud-platform-environments/blob/main/pkg/namespace/namespace.go#L114-L128

Reply:
The namespace package query the hosted_services endpoint to get the list of namespaces. But this one query the cluster. 
We should ideally have 2 functions in the namespace package. "GetAllNamespacesFromCluster" which needs permission to access cluster. Another one "GetAllNamespacesJson" which query the HOODAW hosted services and get a json output

Reply:
Updated the namespace package in this PR: https://github.com/ministryofjustice/cloud-platform-environments/pull/6619